coderd/provisionerjobs.go

```diff
@@ -38,6 +38,37 @@ func (api *API) provisionerJobLogs(rw http.ResponseWriter, r *http.Request, job
 		return
 	}
 
+	// if we are following logs, start the subscription before we query the database, so that we don't miss any logs
+	// between the end of our query and the start of the subscription.  We might get duplicates, so we'll keep track
+	// of processed IDs.
+	var bufferedLogs chan database.ProvisionerJobLog
+	if follow {
+		bl, closeFollow, err := api.followLogs(job.ID)
+		if err != nil {
+			httpapi.Write(rw, http.StatusInternalServerError, httpapi.Response{
+				Message: "Internal error watching provisioner logs.",
+				Detail:  err.Error(),
+			})
+			return
+		}
+		defer closeFollow()
+		bufferedLogs = bl
+
+		// Next query the job itself to see if it is complete.  If so, the historical query to the database will return
+		// the full set of logs.  It's a little sad to have to query the job again, given that our caller definitely
+		// has, but we need to query it *after* we start following the pubsub to avoid a race condition where the job
+		// completes between the prior query and the start of following the pubsub.  A more substantial refactor could
+		// avoid this, but not worth it for one fewer query at this point.
+		job, err = api.Database.GetProvisionerJobByID(r.Context(), job.ID)
+		if err != nil {
+			httpapi.Write(rw, http.StatusInternalServerError, httpapi.Response{
+				Message: "Internal error querying job.",
+				Detail:  err.Error(),
+			})
+			return
+		}
```

Comment:
Understandable :'(

Reply:
I guess technically we could do this query asynchronously and check its result later, since we already know we're going to query logs from the DB regardless of whether the job is already complete. But it's such a lightweight query that it doesn't seem worth the extra complexity.

Reply:
>  But it's such a lightweight query that it doesn't seem worth the extra complexity.
Agree